torch/csrc/inductor/aoti_runtime/model_container.h

```diff
@@ -467,14 +467,29 @@ class AOTInductorModelContainer {
           constants_blob_ptr + constants_internal_offset_[idx];
       void* user_constant_ptr;
       int64_t constant_size;
+      int64_t* stride;
+      int64_t offset;
       aoti_torch_get_data_ptr(tensor, &user_constant_ptr);
       aoti_torch_get_storage_size(tensor, &constant_size);
+      AOTI_TORCH_ERROR_CODE_CHECK(aoti_torch_get_strides(tensor, &stride));
+      AOTI_TORCH_ERROR_CODE_CHECK(
+          aoti_torch_get_storage_offset(tensor, &offset));
+
 #ifdef USE_XPU
       sycl::queue* queue_ptr = nullptr;
       aoti_torch_get_current_sycl_queue((void**)&queue_ptr);
       queue_ptr
           ->memcpy(internal_constants_ptr, user_constant_ptr, constant_size)
           .wait();
+#elif USE_MPS
+      internal_constants_ptr = constants_blob_ptr;
```

Comment:
Why do we need to treat this differently for MPS?

Reply:
I think because metal has unified memory, we can't directly access the contents of the constants ptr through ptr arithmetic. Instead we have to pass the offset around and access it with the metal functions.

Reply:
Other way around: we don't have a unified memory, and storage ptr is not really a pointer but reference to `id<MTLBuffer>` object